### PR TITLE
[Access] update system chunk tx handling in GetTransactionResults API

### PR DESCRIPTION
**Pull request by smnzhu**

### CODE REVIEW

File: engine/access/rpc/backend/backend_transactions.go [hunk 1]
```diff
@@ -342,25 +347,31 @@ func (b *backendTransactions) GetTransactionResultsByBlockID(
...
+		return nil, status.Errorf(codes.Internal, "number of transaction results returned by execution node is more than the number of transactions in the block")
+	}
 
-		i++
+	systemTx, err := blueprints.SystemChunkTransaction(b.chainID.Chain())
+	if err != nil {
+		return nil, fmt.Errorf("could not get system chunk transaction: %w", err)
 	}
+	systemTxResult := resp.TransactionResults[len(resp.TransactionResults)-1]
+	systemTxStatus, err := b.deriveTransactionStatus(nil, true, block)
```
Comment (m4ksio) [inline engine/access/rpc/backend/backend_transactions.go:362]: really just being picky, but shouldn't we pass `systemTx` as a first arg here?

**Review by m4ksio — APPROVED** [review]

File: engine/access/rpc/backend/backend_transactions.go (hunk 1 above)
Comment (smnzhu) [reply engine/access/rpc/backend/backend_transactions.go:362]: True, I can do that

**Review by koko1123 — APPROVED** [review]
looks good to me

**Comment by smnzhu** [comment]
bors merge

**Comment by smnzhu** [comment]
bors cancel

**Comment by smnzhu** [comment]
bors merge

**Pull request merged**